Make the realm icon an img element

jmbowman opened the pull request:
The realm avatar icon on the login and registration pages was
being set as a background image, which could vanish in high
contrast mode in many browsers.  Converted it to an img tag and
verified that it is still styled correctly.  I think the empty
alt attribute (to remove it from the audio description) is
appropriate in this context, since the realm name and description
are already provided immediately afterwards in the page content.

Fixes #4889.

smarx commented:
*Automated message from Dropbox CLA bot*

@jmbowman, it looks like you've already signed the Dropbox CLA. Thanks!

timabbott commented:
@brockwhittaker can you review this one?  I'm not sure why we were using background-image in the first place.

brockwhittaker commented:
This works, the only last piece would be to delete the unnecessary CSS associated with this, which is here: 

```css
.split-view .org-header .avatar,
.register-page-container .org-header .avatar {
    display: inline-block;
    vertical-align: top;

    width: 100px;
    height: 100px;

    background-size: 100% auto;
}
```

Removing the `background-size` attribute.

jmbowman commented:
Removed the `background-size` attribute and rebased, although it looks like some of the Travis jobs are hitting an error installing Twisted (which looks distinctly unrelated to my changes, so I'm guessing that's a problem in the current master branch).

lonerz commented:
@jmbowman seems like you still have some [lint](https://travis-ci.org/zulip/zulip/jobs/238198690) errors however

timabbott commented:
The twisted installer issue definitely isn't your fault; I think it's an upstream dependency issue we'll have to debug

timabbott commented:
OK I fixed the production/twisted installer issue in master.

jmbowman commented:
Do we really want to force empty alt text strings to go through the i18n translation code?  That seems like it should be a special case in the new linting rule, it just wastes CPU cycles otherwise.

showell commented:
For the empty alt text string, can we either a) just eliminate it or b) come up with a sensible value for it?

If neither of the above apply, then, yes, we can probably alter the lint rule.  It seems to me, though, that for accessibility, we want some kind of text in most of our elements.

brockwhittaker commented:
It was `background-image` for a few reasons:

1. Disables dragging without things that don't work cross-browser (eg. `draggable="false"`) or without resorting to JS (`el.ondragstart = () => false`).
2. Regardless of the ratio of the image, it'll always display as 1:1 in the browser so it'll never break content. Then it can be chosen to be displayed cover/contain to maintain ratio while clipping the edges gracefully.
3. Easier to use pseudo-elements on in general, since you cannot attach them to things that don't have the ability to have `textContent` (eg. `<img>`).

jmbowman commented:
From an accessibility perspective, it's important that the image has alt text, even if it's empty.  If excluded, a screen reader will often read out the filename or path of the image instead.  I'd normally go with the name of the realm as the alt text for this kind of image, but that appears as text immediately after the image here, so it sounds redundant to hear it twice.  The linting rule needs to cope with empty alt text regardless, because [empty text is sometimes the most appropriate content for the alt attribute](http://webaim.org/techniques/alttext/#context).

timabbott commented:
I'm on board with changing the lint rule to support the empty string.  Should be a trivial change in `tools/lint`.

timabbott commented:
@umairwaheed maybe tonight you can tweak the lint rule to allow the empty string as alt text?

umairwaheed commented:
Yeah.

umairwaheed commented:
@timabbott, updated the rule in https://github.com/zulip/zulip/pull/5171

timabbott commented:
Merged after merging #5171, thanks @jmbowman!
